src/orbit.c

```diff
@@ -275,3 +256,32 @@ bool orbit_decayed(const orbit_t *orbit)
 	}
 	return has_decayed;
 }
+
+	/* Calculates if a position is eclipsed.  */
+bool is_eclipsed(const double pos[3], const double sol[3], double *depth)
+{
+	double Rho[3], earth[3];
+
+	/* Determine partial eclipse */
+	double sd_earth = ArcSin(xkmper / vec3_length(pos));
+	vec3_sub(sol, pos, Rho);
+	double sd_sun = ArcSin(sr / vec3_length(Rho));
+	vec3_mul_scalar(pos, -1, earth);
+	
+	double delta = ArcCos( vec3_dot(sol, earth) / vec3_length(sol) / vec3_length(earth) );
+	*depth = sd_earth - sd_sun - delta;
+
+	if (sd_earth < sd_sun) return false;
+	else if (*depth >= 0) return true;
+	else return false;
+}
+
+bool orbit_is_eclipsed(const orbit_t *orbit)
```

Comment:
See https://www.kernel.org/doc/Documentation/CodingStyle "Chapter 5: Typedefs". I would suggest that we keep orbit_t typedef and use accessor functions, or change to struct orbit and drop accessor functions. What do you think?